tests/components/maxcube/test_binary_sensor.py

```diff
@@ -0,0 +1,56 @@
+"""Test EQ3 Max! Window Shutters."""
+
+from unittest.mock import MagicMock
+
+from maxcube.device import MaxDevice
+from maxcube.windowshutter import MaxWindowShutter
+
+from homeassistant.components.binary_sensor import DEVICE_CLASS_WINDOW
+from homeassistant.components.maxcube.binary_sensor import setup_platform
+
+from . import mock_cube_handle, mock_hass
+
+
+def test_no_device_setup():
+    """Test a successful setup without any device."""
+    entities = []
+    handler = mock_cube_handle([])
+    setup_platform(mock_hass(handler), None, lambda l: entities.extend(l))
```

Comment:
Please don't interact with integration details in the tests. Use the core to start the tests and assert test state.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
I am still working on it. Hope to have it ready in a few more days

Reply:
Reworked and rebased on top of https://github.com/home-assistant/core/pull/48047
Please, review and merge that CR first.

Reply:
That PR is merged. :+1: